packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -580,19 +587,19 @@ export class ConversationService extends TypedEventEmitter<Events> {
       );
 
       await this.joinByExternalCommit(mlsConversation.qualified_id);
-      return this.getMLS1to1Conversation(otherUserId);
+      return (await this.getMLS1to1Conversation(otherUserId)).conversation;
     }
 
     // If group is not established on backend,
     // we wipe the it locally (in case it exsits in the local store) and try to register it.
     await this.mlsService.wipeConversation(groupId);
 
     try {
-      await this.mlsService.register1to1Conversation(groupId, otherUserId, selfUser);
+      await this.mlsService.register1to1Conversation(groupId, otherUserId, selfUser, public_keys?.removal);
```

Comment:
This is the main change, we can register a conversation with a removal key that comes from a different endpoint